omnikinverter/models.py

```diff
@@ -200,6 +200,40 @@ def from_tcp(data: dict[str, Any]) -> Inverter:
             ),
         )
 
+    @staticmethod
+    def from_cgi(data: str) -> Inverter:
+        """Return Inverter object from the Omnik Inverter response.
+
+        Args:
+            data: The CGI (webscraping) data from the Omnik Inverter.
+
+        Returns:
+            An Inverter object.
+        """
+
+        split_data = data.split(";")
+        if len(split_data) < 7:
+            raise OmnikInverterWrongSourceError(
+                    "Your inverter has no data source from cgi."
+                )      
+
+        def try_parse_float(item):
+            try:
+                return float(item)
+            except:
+                return None
+
+        return Inverter(
+            serial_number=split_data[0],
+            model=split_data[2],
+            firmware=split_data[1],
+            firmware_slave=None,
+            solar_rated_power=None,
+            solar_current_power=try_parse_float(split_data[4]),
+            solar_energy_today=try_parse_float(split_data[5]),
+            solar_energy_total=try_parse_float(split_data[6]),
```

Comment:
How do you know that this is encoded in the seventh field? Your test fixture has the non-float value `d` here?

Reply:
I am going by the variable names the Web UI uses in its request:

```
    set_inner_by_id("webdata_sn", inverterArr[0]);
    set_inner_by_id("webdata_msvn", inverterArr[1]);
    set_inner_by_id("webdata_pv_type", inverterArr[2]);
    set_inner_by_id("webdata_rate_p", inverterArr[3] + "℃");
    set_inner_by_id("webdata_now_p", inverterArr[4] + "W");
    set_inner_by_id("webdata_today_e", inverterArr[5] + "kWh");
    set_inner_by_id("webdata_total_e", inverterArr[6] + "kWh");
    set_inner_by_id("webdata_alarm", inverterArr[7]);
```
They match with the 'old' ones.
The data on my Web UI displays 'd' for that field:
![Screenshot_2022-11-04_19-31-46](https://user-images.githubusercontent.com/915407/199928167-fc43f642-fbd2-4708-940a-90b90a22d40b.png)


That being said, I noticed that webdata_rate_p is wrong as it returns the temperature instead...

Reply:
Looks like they just reused an existing `id=` and changed the unit and label text (with a typo `temprature` :joy:) :)